torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2090,11 +2090,19 @@ XLATensorPtr pow(const XLATensorPtr& input, const at::Scalar& exponent) {
   // optimize
   torch::lazy::Value exponent_node =
       XLAGraphExecutor::Get()->GetIrValueForConstant(exponent, input->shape());
-  return input->CreateFrom(Pow(input->GetIrValue(), exponent_node));
+  torch::lazy::NodePtr node = Pow(input->GetIrValue(), exponent_node);
+  auto* xla_node = dynamic_cast<XlaNode*>(node.get());
+  at::ScalarType dtype =
+      TorchTypeFromXlaType(xla_node->xla_shape().element_type());
+  return input->CreateFrom(node, dtype);
```

Comment:
why do we need to set the logical element type here? It does not impact the HLO graph. I think it is only being used by `XLATensor::dtype()`, 

https://github.com/pytorch/xla/blob/dbd3e33c45e53dc5725a07a3508f0d40d9e7e32e/torch_xla/csrc/tensor.cpp#L159C16-L163

If we don't set it, it will fallback to `xla_node->xla_shape().element_type()` anyway

Reply:
If we don't set it; it will use element type of `input`. So the result will get casted from float to int